pkg/apis/core/validation/validation.go

```diff
@@ -85,6 +85,7 @@ var allowedEphemeralContainerFields = map[string]bool{
 	"TerminationMessagePath":   true,
 	"TerminationMessagePolicy": true,
 	"ImagePullPolicy":          true,
+	"SecurityContext":          true,
```

Comment:
is there any coverage of this field for PodSecurity and PodSecurityPolicy?

Reply:
For PodSecurityPolicy I added tests to [podsecuritypolicy/provider_test.go](https://github.com/kubernetes/kubernetes/pull/99023/files#diff-8a6845b20e9578568a9c8018a7999c46a283ea6c370d436740ee7699a92e4a9d) to run the Container tests on EphemeralContainer as well.

I wasn't following the PSP replacement and didn't know about PodSecurity until you mentioned it. I'm happy to add test coverage there as well once I bring myself up to speed. If the PSP coverage looks sufficient, how do you feel about addressing PodSecurity in a follow-up PR since it's still in alpha?

Reply:
as long as there's double opt-in and we have test coverage for the interaction before either this feature or PodSecurity graduates from alpha, that's ok

Reply:
including test coverage for ephemeral containers is in the PodSecurity KEP for alpha → beta graduation

fwiw, running with PodSecurity enabled, I just verified the PodSecurity enforce level already applies to ephemeral containers as well:

```
FEATURE_GATES=PodSecurity=true,EphemeralContainers=true hack/local-up-cluster.sh 

kubectl label ns default pod-security.kubernetes.io/enforce=restricted

kubectl apply -f ~/snippets/pods/restricted_pod.json 

kubectl debug restricted -it --image=busybox

Defaulting debug container name to debugger-86nkb.
Error from server (Forbidden): allowPrivilegeEscalation != false (container "debugger-86nkb" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "debugger-86nkb" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "debugger-86nkb" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "debugger-86nkb" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
```